tools/main.py

```diff
@@ -150,8 +151,10 @@ def main():
                 if not check_classes_consistency(model.classification_classes,
                                                  dataloader['query'].dataset.classes, strict=False):
                     raise ValueError('Inconsistent classes in evaluation dataset')
+                if args.classes and not check_classes_consistency(classes_map,
+                                                                dataloader['query'].dataset.classes, strict=True):
+                    raise ValueError('Classes provided via --classes argument and in the evaluation dataset are different')
```

Comment:
We should carry out the evaluation if `args.classes` is subset of `dataset.classes` (the same situation if `dataset.classes` is subset of `model.classification_classes`).

Reply:
Ok. May be I misunderstood IK. But why `args.classes` should be a subset of `dataset.classes` and not vice versa?
If `args.classes` are supposed to replace metainfo when using IR, it should be identical to `model.classification_classes` and thus `dataset.classes` is a subset of `args.classes`.

Reply:
My understanding:
- In train mode: `--classes` is subset of `dataset.classes`. If it's passed in args then filter `dataset.classes`.
- In eval mode: `--classes` is subset of both `dataset.classes` which is subset of `model.classes` - if it's passed in args then filter `dataset.classes`.

Reply:
<pre>
In train mode: --classes is subset of dataset.classes. If it's passed in args then filter dataset.classes.
</pre>
Correct

<pre>
In eval mode: --classes is subset of both dataset.classes which is subset of model.classes - if it's passed in args then filter dataset.classes.
</pre>
In eval mode: classes are always taken from snapshot in case of pytorch, from some file near to IR in case of OpenVINO (we can agree on it). You can see my PR: https://github.com/openvinotoolkit/training_extensions/pull/434/files#diff-85117c21b0b255840f45ceb9bafcecf15036c2d95af7e0e45bbe78dd13070218R120

@LeonidBeynenson it's hard to imagine where we can use --classes in eval.py, compres.py and export.py. Is it worth not to pass them?

@evgeny-izutov @sovrasov, we with @LeonidBeynenson agreed to pass extra args everywhere just in case if it might be needed somehow for somealgo.

Reply:
Implemented filtration before training and evaluation . According to the final agreement, in eval mode --classes should be the same as model.classes (if both are provided).